### PR TITLE
Add new typography method in Headings component

### LINKED ISSUE #343: Implement new font method in Headings component
Blocked by #342 

**Is your feature request related to a problem? Please describe.**
World Service has a UX requirement that GEL Types (Canon, Long Primer, Double Pica, etc) should allow for mutable values across services. These values were grouped into 11 different scripts: `arabic`, `arabic-pashto`, `bengali`, `burmese`, `devanagari-and-gurmukhi`, `latin-and-cyrillic`, `latin-with-diacritics`, `no-ascenders-or-descenders`, `sinhalese`, `tamil`, and `thai`.

The `latin-and-cyrillic` script contains the default values GEL for all the Types. The previous constants (GEL_CANON, GEL_LONG_PRIMER, GEL_DOUBLE_PICA, etc) are being deprecated in favour of script-specific methods such as `getCanon()`, `getLongPrimer()` and so on. These methods take the current component's script as an argument.

This issue is part of a wider work to update all existing components to use the new methods, eventually removing the old constants from the codebase.

**Describe the solution you'd like**
* Allow the component to pass the script name and children as a prop.
* Update the component's propTypes and defaultProps to include a prop of `script` that defaults to `latin`
* Import the appropriate getTypeSizes() method for the component's Type Styles (getCannon for Headline, getTrafalgar for Subheading)
* Implement the method above, replacing any references to the deprecated GEL Type constant (GEL_CANON, GEL_TRAFALGAR) in the styled-component CSS string 
* Remove existing imports of the previous GEL Type constant

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

**Additional context**

- [ ] Initially labelled with ["Refinement needed"](https://github.com/bbc/psammead/labels/Refinement%20Needed)

### PR DESCRIPTION
**Pull request by DenisHdz**
Resolves #343

**Overall change:** 
Add support to allow the `Headings` component to use Typography Type Sizes for different scripts

**Code changes:**

- Allow the component to pass the script name a prop
- Add `propTypes` to require a `script` prop
- Remove previous `GEL_CANON` and `GEL_TRAFALGAR` imports
- Import new `getCanon` and `getTrafalgar` methods which make use of the new `getTypeSizes()` method to get the appropriate GEL Type Sizes for the component.
- Import `latin.js` script with the default typography GEL Types Sizes in the stories and unit test
---

- [X] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

File: packages/components/psammead-headings/src/index.jsx [hunk 1]
```diff
@@ -42,5 +44,12 @@ export const SubHeading = styled.h2.attrs(({ text }) => ({
     padding-top: ${GEL_SPACING_QUAD};
   }
   font-weight: 700;
-  ${GEL_TRAFALGAR};
 `;
+
+Headline.defaultProps = {
+  script: objectOf(object).required,
```
Comment (dr3) [inline packages/components/psammead-headings/src/index.jsx:50]: Have raised https://github.com/bbc/psammead/issues/425 to improve this

**Review by dr3 — DISMISSED** [review]
LGTM, Great work 🎉

File: packages/components/psammead-headings/src/index.test.jsx [hunk 2]
```diff
@@ -1,13 +1,18 @@
...
 
 describe('Headline component', () => {
   shouldMatchSnapshot(
     'should render correctly',
     <Headline script={latin}>This is my headline.</Headline>,
   );
+
+  shouldMatchSnapshot(
+    'should render correctly with arabic script typography values',
+    <Headline script={arabic}>This is my headline.</Headline>,
```
Comment (dr3) [inline packages/components/psammead-headings/src/index.test.jsx]: I think this should have arabic words

**Review by dr3 — CHANGES_REQUESTED** [review]

File: packages/components/psammead-headings/src/index.test.jsx (hunk 2 above)
Comment (DenisBBC) [reply packages/components/psammead-headings/src/index.test.jsx]: I have just updated before your review 😄

**Comment by DenisHdz** [comment]
Conflicts fixed 🛠

**Review by j-pendlebury — APPROVED** [review]

**Review by dr3 — APPROVED** [review]

**Comment by jamesbrumpton** [comment]
LGTM. Puedes unirlo.

**Pull request merged**